akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbStore.scala

```diff
@@ -14,21 +14,26 @@ import akka.persistence.journal.WriteJournalBase
 import akka.serialization.SerializationExtension
 import org.iq80.leveldb._
 
+import scala.collection.convert.WrapAsScala.mapAsScalaMap
 import scala.collection.immutable
 import scala.util._
 import scala.concurrent.Future
 import scala.util.control.NonFatal
 import akka.persistence.journal.Tagged
-import com.typesafe.config.{ Config, ConfigFactory }
+import com.typesafe.config.{ Config, ConfigFactory, ConfigObject }
 
 private[persistence] object LeveldbStore {
   val emptyConfig = ConfigFactory.empty()
+
+  def toCompactionIntervalMap(obj: ConfigObject): Map[String, Long] = {
+    obj.unwrapped().map(entry ⇒ (entry._1, java.lang.Long.parseLong(entry._2.toString))).toMap
```

Comment:
perhaps could be something like `obj.unwrapped().mapValues(v => java.long.parseLong(v.toString)))`?

Reply:
According to the scala docs 'mapValues' returns a view instead of a concrete map (confirmed it with a code sample). Hence, should we go with something like this instead?
`obj.unwrapped().transform((_, v) => java.long.parseLong(v.toString))`

Reply:
@raboof I would prefer a concrete collection instead of view, as it is used after every single delete, multiple times

Reply:
yes, we shouldn't use views

Reply:
please use the more explicit version:
```
import scala.collection.JavaConverters._
obj.unwrapped().asScala.map
```

Reply:
Agreed. I will keep the mapping process as is (```map(entry ⇒ (entry._1, java.lang.Long.parseLong(entry._2.toString)))```), since this allows for using both long and string literals in the configuration (i.e. both 4 and "4" will work just fine)